src/object_gui.cpp

```diff
@@ -544,8 +544,37 @@ class BuildObjectWindow : public Window {
 	void OnPlaceObject(Point pt, TileIndex tile) override
 	{
 		ObjectClass *objclass = ObjectClass::Get(_selected_object_class);
```

Comment:
We use only the object spec, so we do not need the object class separately:
```suggestion
		const ObjectSpec *spec = ObjectClass::Get(_selected_object_class)->GetSpec(_selected_object_index);
```